### DISCUSSION TITLE
请问F103实现高速HID设备怎么做到的

### DISCUSSION THREAD

**Discussion by wlfinal** [peng-zhihui/HelloWord-Keyboard#80] [Q&A] [answered]
初学者，刚刚入坑自己随便玩玩。
简介里提到“HID高速设备键盘枚举”，看数据手册里F103支持USB全速，没有高速USB控制器，想知道这个功能怎么实现的？因为我用的ESP32（主要是便宜+蓝牙键盘），如果stm32可以做到，ESP32也可以同样思路尝试解决，或者放弃高速HID。
全速HID只有1ms/次的响应速度，和高速125us/3次差距略大啊（其实用起来感觉不出来，就是感觉这么好的扫描速度别浪费了）

**Comment by peng-zhihui** [comment, marked as answer]
打错了，是全速，F1没有高速PHY
